### PR TITLE
Migrate config-ssh.yaml to Slurm-GCP v6

### PR DESCRIPTION
**Pull request by alyssa-sm**
This PR updates `config-ssh.yaml` to Slurm-GCP v6.

Calling `sinfo` shows the nodes.
```
PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
compute      up   infinite     20  idle~ hpcsmall-comput-[0-19]
debug*       up   infinite      4  idle~ hpcsmall-debugn-[0-3]
```

Manually submitting the job `sbatch -N 1 -p compute --wrap "sleep 10"` successfully completes.
```
JobId=2 JobName=wrap
   UserId=alyssasm_hpctoolkit_joonix_net(1027357380) GroupId=alyssasm_hpctoolkit_joonix_net(1027357380) MCS_label=N/A
   Priority=1 Nice=0 Account=(null) QOS=normal
   JobState=COMPLETED Reason=None Dependency=(null)
   Requeue=1 Restarts=0 BatchFlag=1 Reboot=0 ExitCode=0:0
   RunTime=00:00:10 TimeLimit=UNLIMITED TimeMin=N/A
   SubmitTime=2024-03-13T05:07:30 EligibleTime=2024-03-13T05:07:30
   AccrueTime=2024-03-13T05:07:30
   StartTime=2024-03-13T05:10:59 EndTime=2024-03-13T05:11:09 Deadline=N/A
   SuspendTime=None SecsPreSuspend=0 LastSchedEval=2024-03-13T05:07:31 Scheduler=Main
   Partition=compute AllocNode:Sid=hpcsmall-controller:4157
   ReqNodeList=(null) ExcNodeList=(null)
   NodeList=hpcsmall-comput-0
   BatchHost=hpcsmall-comput-0
   NumNodes=1 NumCPUs=30 NumTasks=1 CPUs/Task=1 ReqB:S:C:T=0:0:*:*
   ReqTRES=cpu=1,mem=7938M,node=1,billing=1
   AllocTRES=cpu=30,mem=238140M,node=1,billing=30
   Socks/Node=* NtasksPerN:B:S:C=0:0:*:* CoreSpec=*
   MinCPUsNode=1 MinMemoryCPU=7938M MinTmpDiskNode=0
   Features=(null) DelayBoot=00:00:00
   OverSubscribe=NO Contiguous=0 Licenses=(null) Network=(null)
   Command=(null)
   WorkDir=/home/alyssasm_hpctoolkit_joonix_net
   StdErr=/home/alyssasm_hpctoolkit_joonix_net/slurm-2.out
   StdIn=/dev/null
   StdOut=/home/alyssasm_hpctoolkit_joonix_net/slurm-2.out
   Power=
```

### CODE REVIEW

File: tools/validate_configs/test_configs/config-ssh.yaml [hunk 1]
```diff
@@ -46,54 +46,53 @@ deployment_groups:
...
     settings:
       partition_name: debug
       exclusive: false # allows nodes to stay up after jobs are done
-      enable_placement: false # the default is: true
       is_default: true
 
   - id: compute_node_group
-    source: community/modules/compute/schedmd-slurm-gcp-v5-node-group
+    source: community/modules/compute/schedmd-slurm-gcp-v6-nodeset
+    use: [network1]
```
Comment (harshthakkar01) [inline tools/validate_configs/test_configs/config-ssh.yaml:68]: Lets update to `network` in the blueprint.

Comment (alyssa-sm) [reply tools/validate_configs/test_configs/config-ssh.yaml]: I replaced `network1` with `network`

**Review by harshthakkar01 — APPROVED** [review]

File: tools/validate_configs/test_configs/config-ssh.yaml [hunk 2]
```diff
@@ -31,12 +31,12 @@ deployment_groups:
   # Source is an embedded resource, denoted by "resources/*" without ./, ../, /
```
Comment (mr0re1) [inline tools/validate_configs/test_configs/config-ssh.yaml:31]: nit. "resource" is deprecated word, replaced by "module". Pls, either update the comment or remove it all together.

Comment (mr0re1) [reply tools/validate_configs/test_configs/config-ssh.yaml:31]: never mind, I've addressed it in #2364

**Pull request merged**